device/common/include/traccc/clusterization/device/impl/reduce_problem_cell.ipp

```diff
@@ -28,7 +28,7 @@ inline void reduce_problem_cell(
     const unsigned int pos = cid + start;
 
     // Load the "reference cell" into a local variable.
-    const cell reference_cell = cells.at(pos);
+    const auto reference_cell = cells.at(pos);
```

Comment:
```suggestion
    const auto& reference_cell = cells.at(pos);
```

Reply:
Proxies are not members of the SoA containers. They are returned by value by `at(...)`. So using a const reference is at best confusing, at worst it may lead to problems. (Though from what I understand, the compiler would still generate valid code with that. It would just confuse the reader I believe.)

Maybe this is the source of our disagreement? :confused: Every time the code creates a proxy, it's not for free. We can hope that the compiler does a good job with optimizing the code. But if the compiler is not smart enough, `container.at(i).foo()` will be slower than `container.foo().at(i)`.